junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestParameterResolver.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.jupiter.params;
+
+import java.lang.reflect.Parameter;
+import java.util.Optional;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.params.converter.ArgumentConverter;
+import org.junit.jupiter.params.converter.ConvertWith;
+import org.junit.jupiter.params.converter.DefaultArgumentConverter;
+import org.junit.jupiter.params.support.AnnotationInitializer;
+import org.junit.platform.commons.util.AnnotationUtils;
+import org.junit.platform.commons.util.ReflectionUtils;
+
+class ParameterizedTestParameterResolver implements ParameterResolver {
+
+	private final Object[] arguments;
+
+	ParameterizedTestParameterResolver(Object[] arguments) {
+		this.arguments = arguments;
+	}
+
+	@Override
+	public boolean supports(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		return parameterContext.getIndex() < arguments.length;
```

Comment:
Yes, for now we would require them to be the first arguments. We could later introduce an annotation, e.g. `@Parameter` with an optional `index` value to mark them explicitly.